StardewArchipelago/GameModifications/Modded/JunimoShopGenerator.cs

```diff
@@ -335,56 +339,90 @@ private static bool IsColor(string itemContext, string color)
+            var item = _stardewItemManager.GetObjectById(itemId);
+            if (isSeed == true && _archipelago.SlotData.Cropsanity == Cropsanity.Shuffled && !_archipelago.HasReceivedItem(item.Name))
             {
                 return;
             }
-            if (itemSeason != null)
+            if (itemSeason != null && !itemSeason.Contains(Game1.currentSeason))
             {
-                if (!itemSeason.Contains(Game1.currentSeason))
-                {
-                    return;
-                }
+                return;
             }
-            AddToJunimoStock(stock, _stardewItemManager.GetItemByName(itemName), color, null);
+            AddToJunimoStock(stock, item, color, null);
         }
 
+        public int[] ExchangeRate(int soldItemValue, int requestedItemValue)
+        {
+            if (IsOnePriceAMultipleOfOther(soldItemValue, requestedItemValue, out var exchangeRate))
+                {
+                    return exchangeRate;
+                }
+            var greatestCommonDivisor = GreatestCommonDivisor(soldItemValue, requestedItemValue);
+            var leastCommonMultiple = soldItemValue * requestedItemValue / greatestCommonDivisor;
+            var soldItemCount = leastCommonMultiple/soldItemValue;
+            var requestedItemCount = leastCommonMultiple/requestedItemValue;
+
+            var applesDiscount = GiveApplesFriendshipDiscount(soldItemCount, requestedItemCount);
+            soldItemCount = applesDiscount[0];
+            requestedItemCount = applesDiscount[1];
 
-        // Lets just say they no currency good cuz Junimo
-        public static int[] ExchangeRate(int soldItem, int wantedItem)
+            var lowestCount = 5; // This is for us to change if we want to move this value around easily in testing
+            var finalCounts = MakeMinimalCountBelowGivenCount(soldItemCount, requestedItemCount, lowestCount);
+            
+            return finalCounts;
+        }
+
+        private bool IsOnePriceAMultipleOfOther(int soldItemValue, int requestedItemValue, out int[] exchangeRate)
         {
-            if (soldItem > wantedItem && soldItem % wantedItem == 0)
-                return new int[2]{1, soldItem / wantedItem};
-            if (soldItem <= wantedItem &&  wantedItem % soldItem == 0)
-                return new int[2]{wantedItem / soldItem, 1};
-            var gcd = Gcd(soldItem, wantedItem);
-            var lcm = soldItem * wantedItem / gcd;
-            var requestCount = lcm/soldItem;
-            var offerCount = lcm/wantedItem;
+            exchangeRate = null;
+            if (soldItemValue > requestedItemValue && soldItemValue % requestedItemValue == 0)    
+                {
+                    exchangeRate = new int[2]{1, soldItemValue / requestedItemValue};
+                    return true;
+                }
+            if (soldItemValue <= requestedItemValue &&  requestedItemValue % soldItemValue == 0)
+                {
```

Comment:
wrong indent. You should really look into enabling the auto formatting of files. I'm sure VSCode has it